queries/nix.scm

```diff
@@ -0,0 +1,146 @@
+;; Statements
+
+[
+    (binding)
+] @statement
+;; This matches assert statements as stand-alone statements. This
+;; is because it's common to have a series of assert statements in a file, which
+;; would otherwise all match as a single statement. See the playground/nix/asserts.nix
+;; file for an example.
+(assert_expression
+    "assert" @statement.start
+    condition: (_)
+    ";" @statement.end
+    body: (_)
+)
+(binding_set) @map.iteration @list.iteration @ifStatement.iteration
+(source_code) @statement.iteration @map.iteration @comment.iteration
+
+;;!! let
+;;!!   a = 1;
+;;!    <*****
+;;!!   b = 2;
+;;!    *****>
+;;!! in a + b
+(let_expression
+    "let"
+    (_) @statement.iteration @value.iteration @name.iteration
+)
+
+;; Conditionals
+
+;;!! key = if a then b else c;
+;;!        ^^^^^^^^^^^^^^^^^^
+(if_expression) @ifStatement @branch.iteration
+;;!! key = if a then b else c;
+;;!        ^^^^
+(if_expression
+    "if" @branch.start
+    condition: (_) @branch.end
+)
+;;!! key = if a then b else c;
+;;!             ^^^^^^
+(if_expression
+    "then" @branch.start
+    consequence: (_) @branch.end
+)
+;;!! key = if a then b else c;
+;;!                    ^^^^^^
```

Comment:
in TS, we don't include the `else` in ternaries (ie the `:`), but we do in full `if` statements 🤔 cc/ @AndreasArvidsson

Reply:
Tested in rust and it seems the else in `let foo = if ... else...` is included for branches.

But rust doesn't have a separate ternary operators (like ? and :), nor does nix.

Reply:
Not sure if there is anything for me to do here, so leaving as open, but if you think this is ok I will mark as resolved.

Reply:
I'm on the fence here. I do often swap branches in ternaries in typescript, which would break here. Otoh when it looks like a big if-else statement this behaviour is more consistent with what we do In typescript. Idk cc/ @AndreasArvidsson